File: cogs/emoji.py
import discord
import asyncio
from discord.ext import commands

import config



class emoji(commands.Cog):

  def __init__(self, client):
    self.client = client


  client = commands.Bot(command_prefix = config.prefixList, case_insensitive=True, intents = discord.Intents().all())





  @commands.command()
  async def getemoji(self, ctx, server_id=None):

    if server_id is None:
      server = self.client.get_guild(ctx.guild.id)
    else:
      server = self.client.get_guild(int(server_id))
    
    pages = []
    if server != None:      #bot in server
      n = 0
      var = ''
      num = 0

      for emoji in server.emojis:
        num += 1

        if n == 20:
          pages.append(var.replace("\\", ""))
          var = ''
          n = 0

        var += f"{emoji} - `\{emoji}` \n"

        n += 1

      #after full pages
      if n >= 1:
          pages.append(var.replace("\\", ""))

      page = 0
      
      if server.icon:
        server_icon = server.icon_url
      else:
        server_icon = self.client.user.avatar_url

      eembed = discord.Embed(
        title = f"Emoji list for {server}",
        description = pages[page],
        color = config.hina_color
      )
      eembed.set_author(name=f"Page ({page + 1} / {len(pages)})", icon_url = server_icon)

      timedOut = False
      emoji_embed = await ctx.reply(embed = eembed)
      await emoji_embed.add_reaction('a<a:Aqua_left:879530551038603264>')
      await emoji_embed.add_reaction('a<a:Aqua_right:879530551881637930>')

      while timedOut == False: 

        try:
          def check(reaction, user):
            return reaction.message.id == emoji_embed.id and user != self.client.user #check added reactions

          reaction, user = await self.client.wait_for('reaction_add', timeout = 60.0, check = check)
          

          if reaction.emoji.id == 879530551038603264 and page > 0:  #detect flip left && not the first page

            page -= 1
            eembed = discord.Embed(
              title = f"Emoji list for {server} [{num}]",
              description = pages[page],
              color = config.hina_color
            )
            eembed.set_author(name=f"Page ({page + 1} / {len(pages)})", icon_url = server_icon)

            await emoji_embed.edit(embed = eembed)          #flip 
            await emoji_embed.remove_reaction('a<a:Aqua_left:879530551038603264>', user)


          if reaction.emoji.id == 879530551881637930 and page < len(pages) -1:  #detect flip right && not the last page

            page += 1
            eembed = discord.Embed(
              title = f"Emoji list for {server}",
              description = pages[page],
              color = config.hina_color
            )
            eembed.set_author(name=f"Page ({page + 1} / {len(pages)})", icon_url = server_icon)

            await emoji_embed.edit(embed = eembed)          #flip
            await emoji_embed.remove_reaction('a<a:Aqua_right:879530551881637930>', user)

        except asyncio.TimeoutError:
          timedOut = True
          await emoji_embed.clear_reactions()

    else:                   #bot not in server
      await ctx.reply("I'm not in the server!")





  @commands.command()
  async def usemoji(self, ctx, given_id=None):

    if given_id == None:
      await ctx.send("Please provide an emoji_id!")
      return
    
    emoji_ID = f"<{given_id}>"

    if ctx.author.nick:     #if user has nickname
      webhook = await ctx.channel.create_webhook(name=ctx.author.nick)
      await webhook.send(str(emoji_ID), username=ctx.author.nick, avatar_url=ctx.author.avatar_url)
      await ctx.message.delete()
      await webhook.delete()
    
    else:
      webhook = await ctx.channel.create_webhook(name=ctx.author.name)
      await webhook.send(str(emoji_ID), username=ctx.author.name, avatar_url=ctx.author.avatar_url)
      await ctx.message.delete()
      await webhook.delete()



  @commands.command()
  async def reactemoji(self, ctx, msg_id, emoji_id):

    msg = await ctx.fetch_message(msg_id)   #get msg user wants to react to

    emoji_id = '<' + emoji_id + '>'         #complete the emoji id
    await msg.add_reaction(emoji_id)        #react
    await ctx.message.delete()                      #remove user command





def setup(client):
  client.add_cog(emoji(client))

File: commands/emoji.js
import { MessageEmbed } from 'discord.js';

import { hinaColor, okEmoji } from '../res/config.js';
import { paginator } from '../utils/paginator.js';


export const commands = [

    {
        name: 'getemoji',
        aliases: [],
        description: 'get all server emoji(s)',
        async execute(client, msg, args) {

            if (!msg.guild.emojis.cache.size) return await msg.reply('The server doesn\'t have any emoji!');
            
            const emojiAmount = msg.guild.emojis.cache.size;
            const pageAmount = Math.ceil(emojiAmount / 20);
            let pages = [];
            let page = '';
            let _ = 0;
            for (const [id, emoji] of msg.guild.emojis.cache.entries()) {

                page += `${emoji}- \`${emoji}\`\n`;
                _++;
                
                if (_ === 20) {
                    const embed = new MessageEmbed()
                        .setColor(hinaColor)
                        .setAuthor({name: `${client.user.username} Page ${pages.length + 1} / ${pageAmount}`, iconURL: client.user.displayAvatarURL({size: 4096})})
                        .setTitle(`Emoji Id(s) for ${msg.guild.name} [${msg.guild.emojis.cache.size}]`)
                        .setDescription(page)
                        .setFooter({text: `Requested by: ${msg.author.tag}`, iconURL: msg.author.displayAvatarURL({size: 4096, dynamic: true})})
                        .setTimestamp();
                    
                    pages.push(embed);
                    page = '';
                    _ = 0;
                };
            };
            if (_)  {
                const embed = new MessageEmbed()
                    .setColor(hinaColor)
                    .setAuthor({name: `${client.user.username} Page ${pages.length + 1} / ${pageAmount}`, iconURL: client.user.displayAvatarURL({size: 4096})})
                    .setTitle(`Emoji Id(s) for ${msg.guild.name} [${msg.guild.emojis.cache.size}]`)
                    .setDescription(page)
                    .setFooter({text: `Requested by: ${msg.author.tag}`, iconURL: msg.author.displayAvatarURL({size: 4096, dynamic: true})})
                    .setTimestamp();

                pages.push(embed);
            }

            await paginator(msg, pages, 120_000);
        }
    },



    {
        name: 'usemoji',
        aliases: [],
        description: 'send the emoji as you!',
        async execute(client, msg, args) {

            const emojiRegex = /^a?:.+:([0-9]{18})$/;

            if (!emojiRegex.test(args[0])) return await msg.reply('Invalid emoji id! Please make sure to copy the whole emoji id without the angle brackets.');
            else {
                const webhook = await msg.channel.createWebhook(msg.member.displayName, {
                    avatar: msg.author.displayAvatarURL({size: 4096, dynamic: true}),
                });

                await webhook.send(`<${args}>`);
                await webhook.delete();
                await msg.delete();
            };
        }
    },



    {
        name: 'reactemoji',
        aliases: [],
        description: 'react to messages using the emoji.',
        async execute(client, msg, args) {

            if (!msg.reference) return await msg.reply('Please reply to the message you want to react to while using the command!');
            if (!args) return await msg.reply('Please provide the emoji id.');

            const theMsg = await msg.fetchReference();
            
            try { 
                await theMsg.react(args[0]);
                await msg.delete();
            }
            catch { await msg.reply(`Invalid emoji id: ${args[0]}`)};
        }
    },
];